editor/ok_interface.py

```diff
@@ -93,116 +30,167 @@ def write(self, message):
     def flush(self):
         sys.__stdout__.flush()
 
-
-def run_tests():
-    LOGGING_FORMAT = '%(levelname)s  | %(filename)s:%(lineno)d | %(message)s'
-    logging.basicConfig(format=LOGGING_FORMAT)
-    log = logging.getLogger('client')  # Get top-level logger
-
-    # noinspection PyUnresolvedReferences
-    from client.cli.ok import parse_input
-    log.setLevel(logging.ERROR)
-
-    args = parse_input(["--all", "--verbose", "--local"])
-
+def capture_output(console, lines):
     old_stdout = sys.stdout
     sys.stdout = out = PrintCapture()
+    result = console._interpret_lines(lines)
+    sys.stdout = old_stdout
+    return result, out.log
 
-    assign = assignment.load_assignment(None, args)
+class PromptOutput(metaclass=ABCMeta):
+    @abstractmethod
+    def represenation(self):
+        pass
 
-    msgs = messages.Messages()
-    for name, proto in assign.protocol_map.items():
-        log.info('Execute {}.run()'.format(name))
-        proto.run(msgs)
+@dataclass
+class AreDifferent(PromptOutput):
+    prompt: str
+    expected: str
+    actual: str
+    def represenation(self):
+        return "{prompt}\n{expected}\n{actual}".format(
+            prompt=self.prompt,
+            expected=pad("; expected: ", ";", self.expected),
+            actual  =pad("; actual  : ", ";", self.actual)
```

Comment:
Maybe capitalizing "expected", "actual" would look better? Not sure though.

Reply:
I think it would look a little out of place since you don't do capitalization in scheme, really.

Reply:
But we then get comments like
`; actual  : SchemeError: unknown identifier: accumulate`,
where part of the comment is capitalized and part of it isn't.

Reply:
ah ok that makes sense. Fixed in commit.